src/Generation/GapicClientV2Generator.php

```diff
@@ -560,7 +561,9 @@ private function construct(): PhpClassMember
             ->withParams($optionsParam)
             ->withAccess(Access::PUBLIC)
             ->withBody(AST::block(
-                Ast::assign($clientOptions, AST::call(AST::THIS, $buildClientOptions)($options)),
+                Ast::assign($clientOptions, AST::call(AST::THIS, $buildClientOptions)(
+                    EmulatorSupportGenerator::generateEmulatorOptionsIfRequired($this->serviceDetails, $options)
+                )),
```

Comment:
I would make this simpler/easier to interpret by making it a separate line such that the generated code only _adds_ a line rather than _modifies_ an existing one. The generated code could instead be

```php
$clientOptions = $this->buildClientOptions($options); // existing, unchanged
$clientOptions = $clientOptions + $this->getDefaultEmulatorConfig(); // new, only added when configured
```

You may need to add the `Ast::assign($clientOptions...` and such to the `generateEmulatorOptions` function so its completely standalone.

Reply:
Sounds good, Changed, thank you!